sogs/model/room.py

```diff
@@ -979,6 +1042,207 @@ def attachments_size(self):
             "SELECT COUNT(*), COALESCE(SUM(size), 0) FROM files WHERE room = :r", r=self.id
         ).first()[0:2]
 
+    def get_reactions(
+        self,
+        message_ids: list[int],
+        user: Optional[User] = None,
+        *,
+        reactors: int = 0,
+        session_ids: bool = True,
+    ):
+        """
+        Returns reaction information for the given messages.  Returns a dict of message id ->
+        reaction info for any messages in the given list that have reactions.  The reaction info is
+        itself a dict where keys are reaction strings and values are a dict containing keys:
+
+        - `"count"` -- the total number of this reaction
+        - `"you"` -- boolean set to true if `user` is given and the user has applied this reaction;
+          omitted otherwise.
+        - `"reactors"` -- if the input parameter `reactors` is greater than 0 then this is a list of
```

Comment:
would having reactions paginated make more sense vs just a limit?

Reply:
Not really; the intended use case is that you get the reaction, and for each one, get the first N users who reacted to show a message such as

"Jon, 051234..., 05767195... and 30 more"

and if you *really* need the entire list (which clients usually won't, but a bot might), go fetch it with the dedicated reactors endpoint.

Reply:
I don't know if this endpoint is even going to be used much; it's mainly there for debugging and, possibly, for a bot that needs to sync up reactions -- but I think even that latter case can be better addressed by something in the bot zmq api.

Reply:
Also: paginations suck for something like this because, typically, you're going to get all of them and will end up doing a `LIMIT x OFFSET y`, and walk your way through all the pages, which just makes everyone slower.